File: src/java/WebNinja/src/main/java/webninja/Issue.java
package webninja;

public class Issue {
    private final String title;
    private final String severity;

    public Issue(String severity, String title) {
        this.severity = severity;
        this.title = title;
    }

    public String getTitle() {
        return title;
    }

    public String getSeverity() {
        return severity;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;

        Issue issue = (Issue) o;

        if (severity != null ? !severity.equals(issue.severity) : issue.severity != null) return false;
        if (title != null ? !title.equals(issue.title) : issue.title != null) return false;

        return true;
    }

    @Override
    public int hashCode() {
        int result = title != null ? title.hashCode() : 0;
        result = 31 * result + (severity != null ? severity.hashCode() : 0);
        return result;
    }
}

File: src/dotnet/WebNinja/WatiNinja/watininja/Issue.cs
namespace WatiNinja.watininja
{
    public class Issue
    {
//START:class
public Issue(string title, string severity) {
    Title = title;
    Severity = severity;
}
public string Title { get; private set; }
public string Severity { get; private set; }
public bool Equals(Issue other) {
    if (ReferenceEquals(null, other)) return false;
    if (ReferenceEquals(this, other)) return true;
    return Equals(other.Title, Title) && Equals(other.Severity, Severity);
}
public override bool Equals(object obj) {
    if (ReferenceEquals(null, obj)) return false;
    if (ReferenceEquals(this, obj)) return true;
    if (obj.GetType() != typeof(Issue)) return false;
    return Equals((Issue)obj);
}
public override int GetHashCode() {
    unchecked {
        return ((Title != null ? Title.GetHashCode() : 0) * 397) ^ 
				(Severity != null ? Severity.GetHashCode() : 0);
    }
}
//END:class

    }

}
